### PR TITLE
[UI] polished navbar design with logo and grid styling

### PR DESCRIPTION
**Pull request by cwang12197**
**[DESCRIPTION]**
- polished the header/navbar ui design with css grid styling, logo, title, and customized search bar 
Screen Shot 2022-07-25 at 1 16 54 AM

### CODE REVIEW

**Comment by cwang12197** [comment]
**[DESCRIPTION]**

- able to display products that have been added to the basket in the "basket" section 
- alerts user if product selected is their own product (may change this to prevent it from showing in general)
- prevents users from adding the same product twice by only adding unique user ids

**[WORKING ON]**

- formatting the UI
- handling a delete button to remove it from the basket 
- adding an "on hold" which disables the product button for other users and adds an "on hold" alert

File: food-waste-ui/src/components/Basket/Basket.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  const [basket, setBasket] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(false);
+  const currentUserId = localStorage.getItem("current_user_id");
+
+  React.useEffect(() => {
+    setIsLoading(true);
+    axios.get("http://localhost:3001/makeapost").then((response) => {
+      let allProducts = response.data.products;
+      let allProductsInBasket = [];
+      for (let i = 0; i < allProducts.length; i++) {
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Basket/Basket.jsx:16]: Is there a way we can do this without a nested for loop? Maybe storing another data struct? 

Is allProducts all the products the market place has within a specific radius? How would this work if we scale larger and many many products are available for users to choose from?

File: food-waste-ui/src/components/Basket/Basket.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if (allProducts[i].basket) {
+          for (let j = 0; j < allProducts[i].basket.length; j++) {
+            if (allProducts[i].basket[j] == currentUserId) {
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Basket/Basket.jsx:19]: Naming is confusing, why is this checking if it == currentUserId?

File: food-waste-ui/src/components/Map/Map.jsx [hunk 3]
```diff
@@ -78,9 +77,9 @@ export default function Map({
     setDestination([]);
   };
 
-  if (!isLoaded) {
-    return <Skeleton />;
-  }
+  // if (!isLoaded) {
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Map/Map.jsx:80]: nit: remove comments

File: food-waste-ui/src/components/Map/Map.jsx [hunk 4]
```diff
@@ -6,15 +6,14 @@ import {
...
-import Skeleton from "react-loading-skeleton";
-import "react-loading-skeleton/dist/skeleton.css";
 import { FaLocationArrow } from "react-icons/fa";
 import Button from "@mui/material/Button";
 import "./Map.css";
 import { IconButton } from "@mui/material";
 import MapMarkers from "../MapMarkers/MapMarkers";
 import { Link } from "react-router-dom";
 import Geocoder from "react-native-geocoding";
+// import Loading from "../Loading/Loading";
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Map/Map.jsx:16]: nit: remove comments

File: food-waste-express-api/routes/home.js [hunk 5]
```diff
@@ -13,4 +13,21 @@ router.get('/:objectId', async (req, res) => {
...
 })
+
+router.post('/:objectId', async (req, res) => {
+    const productId = req.params.objectId;
+    let currentUserId = req.headers["current_user_id"]
+    try {
+        const Product = Parse.Object.extend("Products")
+        const query = new Parse.Query(Product)
+        query.get(productId).then((product) => {
+            product.addUnique("basket", currentUserId)
```
Comment (jennyhan1999) [inline food-waste-express-api/routes/home.js:24]: Could you go over how L20-27 works?

File: food-waste-ui/src/components/Loading/Loading.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import * as React from "react";
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Loading/Loading.jsx:1]: nice!

File: food-waste-ui/src/components/Loading/Loading.jsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { motion } from "framer-motion";
+
+("use strict");
+
+export default function Loading() {
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Loading/Loading.jsx:6]: nice!

File: food-waste-ui/src/components/Navbar/Navbar.jsx [hunk 8]
```diff
@@ -39,51 +40,64 @@ export default function Navbar({
...
+        <input
+          className="search-input"
+          type="text"
+          placeholder="Search for an item"
+          onChange={(e) => handleSearchChange(e)}
+        />
+        <SearchIcon className="search-icon" />
+      </div>
+      <div className="navbar-menu-items">
+        {screenWidth > sideBarWidth &&
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Navbar/Navbar.jsx:64]: Do you handle resizing the website window size? What about the mobile view?

File: food-waste-ui/src/components/Basket/Basket.jsx (hunk 1 above)
Comment (jennyhan1999) [inline food-waste-ui/src/components/Basket/Basket.jsx:16]: Is there a way we can do this without a nested for loop? Maybe storing another data struct? 

Is allProducts all the products the market place has within a specific radius? How would this work if we scale larger and many many products are available for users to choose from?

File: food-waste-ui/src/components/Basket/Basket.jsx (hunk 2 above)
Comment (jennyhan1999) [inline food-waste-ui/src/components/Basket/Basket.jsx:19]: Naming is confusing, why is this checking if it == currentUserId? And what is allProducts[i].basket representing?

File: food-waste-ui/src/components/Map/Map.jsx (hunk 4 above)
Comment (jennyhan1999) [inline food-waste-ui/src/components/Map/Map.jsx:16]: nit: remove comments

File: food-waste-ui/src/components/Map/Map.jsx (hunk 3 above)
Comment (jennyhan1999) [inline food-waste-ui/src/components/Map/Map.jsx:80]: nit: remove comments

File: food-waste-express-api/routes/home.js [hunk 9]
```diff
@@ -13,4 +13,21 @@ router.get('/:objectId', async (req, res) => {
         res.send({err})
     }
 })
+
+router.post('/:objectId', async (req, res) => {
+    const productId = req.params.objectId;
+    let currentUserId = req.headers["current_user_id"]
+    try {
```
Comment (jennyhan1999) [inline food-waste-express-api/routes/home.js:20]: Could you go over how L20-27 works?

File: food-waste-express-api/routes/home.js (hunk 5 above)
Comment (jennyhan1999) [inline food-waste-express-api/routes/home.js:24]: If this is casing on 'basket' maybe you can make this a global variable

**Review by jennyhan1999 — COMMENTED** [review]
It seems like you are doing some basket state handling as well but the commit title says these are just UI changes. You could separate this PR into two PRs for clarity

**Comment by cwang12197** [comment]
**[DESCRIPTION]** 

- added loading states when fetching data 
- added a remove icon in basket 

**[WORKING ON]**

- creating the onHold component

**Comment by cwang12197** [comment]
**[DESCRIPTION]**

- set products on hold and delete them from the market home page 

**[WORKING ON]**

- formatting the on hold 
- being able to remove items from on hold

**Review by jennyhan1999 — COMMENTED** [review]
LGTM!

**Review by jennyhan1999 — COMMENTED** [review]
nice!

File: food-waste-express-api/routes/makeapost.js [hunk 10]
```diff
@@ -33,8 +33,14 @@ router.get('/', async (req, res) => {
     try {
         const query = new Parse.Query("Products")
         query.descending("createdAt")
-        products = await query.find()
-  
+        allProducts = await query.find()
+        let products = []
+
+        allProducts.map((product) => {
+            if (!product.get("placedOnHoldBy")) {
```
Comment (jennyhan1999) [inline food-waste-express-api/routes/makeapost.js:40]: If you are using this input a lot of places maybe we can consider making it a global variable

File: food-waste-ui/src/components/Basket/Basket.jsx [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+  };
+
+  React.useEffect(() => {
+    setIsLoading(true);
+    axios.get("http://localhost:3001/makeapost").then((response) => {
+      let allProducts = response.data.products;
+      let allProductsInBasket = [];
+
+      allProducts.map((product) => {
+        if (product.basket && product.basket.includes(currentUserId)) {
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Basket/Basket.jsx:47]: I feel this relationship is a little confusing naming wise

**Comment by jennyhan1999** [comment]
> 

Lets keep the PR comments simple:

##TITLE
Short title

##SUMMARY
Description of every change this PR has

##TEST PLAN
Unit test, screenshots, videos proving that this code has made the changes stated in the summary

**Comment by cwang12197** [comment]
##CREATION OF BASKET CLASS

##SUMMARY

-  redid basket component by storing products in a separate data structure in Parse
- added functions to fetch product and basket query for code modularity

**Comment by cwang12197** [comment]
REMOVAL OF PRODUCT FROM BASKET ON CLICK 

- removed product from database on click 
- cleaned up code (removed console.logs and //)

File: food-waste-express-api/routes/home.js [hunk 12]
```diff
@@ -21,7 +21,6 @@ router.get("/addtobasket", async (req, res) => {
     try {
         let userBasket = await getUserBasket(currentUserId)
         let userBasketOfIds = await userBasket.get("basketOfProductId") //returns first instance
-        console.log("userBasketOfIds", userBasketOfIds)
```
Comment (jennyhan1999) [inline food-waste-express-api/routes/home.js]: Good to clean up these console.logs but please don't include them in your commits. Each commit should be free of console.logs + dead code + minimal comments. This makes it easier to review

File: food-waste-ui/src/components/BasketCard/BasketCard.jsx [hunk 13]
```diff
@@ -1,14 +1,30 @@
...
 import { IconButton, TrashIcon } from "evergreen-ui";
 import "./BasketCard.css";
+import { reverseGeoCodeAddress } from "../../constants/geoCodesToAddress";
+import { useEffect } from "react";
 
 export default function BasketCard({ product, handleRemoveItemFromBasket }) {
+  const [location, setLocation] = React.useState("");
+
+  useEffect(() => {
+    console.log("product location", product.location);
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/BasketCard/BasketCard.jsx:11]: cleanup

File: food-waste-ui/src/components/Map/Map.jsx [hunk 14]
```diff
@@ -84,23 +85,17 @@ export default function Map({
...
 
   function reverseGeoCodeOriginAddress(input) {
-    Geocoder.init(import.meta.env.VITE_GOOGLE_API_KEY, { language: "en" });
-    Geocoder.from(input)
-      .then((res) => {
-        var addressComponent = res.results[0].formatted_address;
-        setOrigin(addressComponent);
-      })
-      .catch((error) => console.warn(error));
+    if (!input) return;
```
Comment (jennyhan1999) [inline food-waste-ui/src/components/Map/Map.jsx:88]: Great job modularizing your code and creating appropriate helper functions!

**Comment by cwang12197** [comment]
TEST PLAN

This shows the removal of the product in the basket as well as the address (changed from geolocation to an actual location):


https://user-images.githubusercontent.com/68706299/181657517-5df42845-ca7b-486c-8b53-36b0b898e99b.mov

**Comment by cwang12197** [comment]
DISPLAY OF PRODUCTS ON HOLD

- able to show products user has put on hold in the "My Products on Hold" tab
- deletes products off market after putting on hold 
- clears basket after put on hold 

TEST PLAN

- compiles successfully with on hold products hidden on market 
Screen Shot 2022-07-29 at 2 07 37 AM
Screen Shot 2022-07-29 at 2 08 57 AM

**Pull request merged**